ORG-187 Add documentation for Organization Domains and <OrganizationList />

mzhong9723 opened the pull request:
ORG-187

chanioxaris commented on docs/organizations/overview.mdx:
```diff
@@ -33,6 +33,8 @@ All members of an organization, regardless of their role, can view information a
 
 Administrators can [invite other users to join the organization](/docs/organizations/inviting-users). An invitation email is sent out, and organization invitations support adding existing users of your application, or new ones. New users can register once they accept the invitation.
 
+Administrators can also use [verified domains](/docs/organizations/verified-domains) to enhance how users can join organizations. Once a verified domain is added to an organization, administrators can choose between different enrollment modes: automatic invitations and automatic suggestions.
```
Maybe don't list the enrollment modes here as it's high likely that we will forget to also update this copyright

panteliselef replied:
Like this ?

```suggestion
Administrators can also use [verified domains](/docs/organizations/verified-domains) to enhance how users can join organizations. Once a verified domain is added to an organization, administrators can choose between different enrollment modes that will determine how new members will be added.
```

chanioxaris replied:
Yeah i believe this is better and less error prone for the future

mzhong9723 replied:
makes sense, i will update this

chanioxaris commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+
+Domains can be added and verified under an organization by administrators. In order to enable this feature, make sure to enable verified domains in the dashboard's _Organization Settings > Verified domains_. The only parameter needed to to create an organization domain is the `name`.
+
+Domains can be verified through an email verfication code sent to an email that uses the domain. If the administrator adding the domain already has a verified email using that domain in their account, the domain will be automatically verified.
```
```suggestion
Domains can be verified through an email verfication code sent to an email that matches the domain. If the administrator adding the domain already has a verified email using that domain in their account, the domain will be automatically verified.
```

panteliselef commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,169 @@
+description: The `<OrganizationList />` component is used to display organization related memberships, invitations, and suggestions for the user.
+---
+
+# `<OrganizationList />`
+
+<Images
+  alt="Organization List Example"
+  width={992}
+  height={568}
+  src="/docs/images/ui-components/component-org_list.svg"
+/>
```
@mzhong9723  Does this image exist ? I couldn't get it to work locally

mzhong9723 replied:
Added image as a PNG, not sure if we should be using a SVG to be consistent - if so, do we usually export those from Figma?

alexisintech replied:
Left a review comment, but these should be optimized, preferably as WebP.
I use Squoosh, and the default WebP settings.

alexisintech commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,169 @@
+---
+title: <OrganizationList />
+description: The `<OrganizationList />` component is used to display organization related memberships, invitations, and suggestions for the user.
```
We can remove MDX from the SEO description.

```suggestion
description: The <OrganizationList /> component is used to display organization related memberships, invitations, and suggestions for the user.
```

alexisintech commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,169 @@
+
+## Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+  <Tab>
+    <CodeBlockTabs options={["App Router", "Pages Router"]}>
```
Let's use CodeBlockTabs instead of Tabs, and we can have the options be {["Next.js App Router", "Next.js Pages Router", "React", "Remix"]}

Tabs should be more for complex content.

alexisintech commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,169 @@
+
+### Force Organizations
+
+For specific applications, we want to prohibit people from using their personal accounts as they need to be part of an organization.
+Use <code>{"<OrganizationList/>"}</code> with <code>{"hidePeronsal"}</code> in order to guide your user to join or create an organization in order to continue.
```
```suggestion
If you would like to prohibit people from using their personal accounts and force them to be part of an organization, the `hidePersonal` property forces your user to join or create an organization in order to continue.
```

alexisintech commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,169 @@
+
+All props below are optional.
+
+<Tables
```
This table does not require complex use of JSX components, so let's use a markdown table.

| Name | Type | Description |
| --- | --- | --- |
| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether `OrganizationList` will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations.<br />Defaults to `false`.

etc etc etc

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+---
+title: Verified Domains
+description: Learn about how to use verified domains for organizations
```
```suggestion
description: Learn about how to use verified domains for organizations.
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+description: Learn about how to use verified domains for organizations
+---
+
+# Creating and Verifying a Domain
```
Sentence-case for titles.

```suggestion
# Creating and verifying a domain
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+
+# Creating and Verifying a Domain
+
+Domains can be added and verified under an organization by administrators. In order to enable this feature, make sure to enable verified domains in the dashboard's _Organization Settings > Verified domains_. The only parameter needed to to create an organization domain is the `name`.
```
```suggestion
Domains can be added and verified under an organization by an administrator. In order to enable this feature, make sure to enable verified domains in the **Verified domains** section of the [Organizations Settings](https://dashboard.clerk.com/last-active?path=organizations-settings) in the Clerk Dashboard.
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+
+Domains can be verified through an email verfication code sent to an email that matches the domain. If the administrator adding the domain already has a verified email using that domain in their account, the domain will be automatically verified.
+
+An instance may only have one verified domain of same name, and an organization may only have one domain of the same name (verified or unverified).
```
```suggestion
An application instance may only have one verified domain of the same name, and an organization may only have one domain of the same name (verified or unverified).
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+
+An instance may only have one verified domain of same name, and an organization may only have one domain of the same name (verified or unverified).
+
+# Enrollment Modes
```
```suggestion
## Enrollment modes
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+
+Once a domain is created and verified, an administrator can choose between two enrollment modes to enable new ways for users to join organizations.
+
+- `automatic invitations` - During sign up, a user will receive an invitation for the organization if their email's domain matches the verified domain. The user will join the organization if they accept the automatic invitation.
```
```suggestion
- `automatic_invitation` - During sign-up, a user will receive an invitation for the organization if their email's domain matches the verified domain. The user will join the organization if they accept the automatic invitation.
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+Once a domain is created and verified, an administrator can choose between two enrollment modes to enable new ways for users to join organizations.
+
+- `automatic invitations` - During sign up, a user will receive an invitation for the organization if their email's domain matches the verified domain. The user will join the organization if they accept the automatic invitation.
+- `automatic suggestions` - During sign up, a user will receive a suggestion for the organization if their email's domain matches the verified domain. The user can request to join, and an administrator must accept this request before the user joins the organization.
```
```suggestion
- `automatic_suggestions` - During sign-up, a user will receive a suggestion for the organization if their email's domain matches the verified domain. The user can request to join, and an administrator must accept this request before the user can join the organization.
```

alexisintech commented on docs/organizations/verified-domains.mdx:
```diff
@@ -0,0 +1,37 @@
+- `automatic invitations` - During sign up, a user will receive an invitation for the organization if their email's domain matches the verified domain. The user will join the organization if they accept the automatic invitation.
+- `automatic suggestions` - During sign up, a user will receive a suggestion for the organization if their email's domain matches the verified domain. The user can request to join, and an administrator must accept this request before the user joins the organization.
+
+# Usage
```
```suggestion
### Usage
```

alexisintech commented on docs/organizations/verified-domains.mdx:
````diff
@@ -0,0 +1,37 @@
+# Usage
+
+```tsx copy
+const { organization, domains } = useOrganization();
````
This `domains` attribute is not documented at all. And neither are its properties such as `prepareAfiilitationVerification`, `attemptAffiliationVerification`, etc.
https://clerk.com/docs/references/react/use-organization

```suggestion
const { organization, domains } = useOrganization();
```

mzhong9723 replied:
Thanks for the catch! Are you able to give me access to the Sanity docs so I can make those updates? If you prefer, I can also open a PR to migrate those docs to MDX, along with the updated `domains` attribute docs.

alexisintech commented on public/images/ui-components/component-org_list.png:
This png needs to optimized. Use Squoosh, and choose WebP and the default settings for WebP are fine.

panteliselef replied:
btw how come we don't have next/Image doing that for us ?

alexisintech replied:
we do, the `<Images />` component is a Next `<Image />`. Compressing to webp saves file size in our repo.

mzhong9723 commented:
Thanks for the thorough review @alexisintech! Addressed all your comments, updated to use a webp image, and left a comment about getting access to Sanity to edit or migrate the `useOrganization` docs.

alexisintech commented:
> Thanks for the thorough review @alexisintech! Addressed all your comments, updated to use a webp image, and left a comment about getting access to Sanity to edit or migrate the `useOrganization` docs.

I can migrate that page today, and I will send you the PR so that you can add to the branch directly.

alexisintech commented:
Migrated our useOrganization docs page to MDX.
Feel free to pull the branch and add the new information!
https://github.com/clerkinc/clerk-docs/pull/332

panteliselef commented:
> Migrated our useOrganization docs page to MDX. Feel free to pull the branch and add the new information! #332

Seems good for now. The API mentioned there is deprecated. And probably early next week I will be updating that page.

panteliselef commented on docs/components/organization/organization-list.mdx:
```diff
@@ -0,0 +1,102 @@
+| `apperance` | `object` | Controls the overall look and feel.
+| `afterCreateOrganizationUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after creating a new organization.<br />Defaults to `undefined`.
+| `afterSelectOrganizationUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after selecting an organization.<br />Defaults to `undefined`.
+| `afterSelectPersonalUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after selecting the personal account.<br />Defaults to `undefined`.
```
the above did not work, i went with 
```suggestion
| `afterSelectPersonalUrl` | <code>((org: [Organization][org-ref]) => string)</code> \| `string` | Full URL or path to navigate after selecting the personal account.<br />Defaults to `undefined`.
```

alexisintech commented:
Compared to the images on the other organization components, I think we need help creating the image for this component. It looks like they were done by Corbin and James with an SVG - I would reach out to one of them to ask how to replicate.

alexisintech reviewed (approved):
WOOOOOO 😸💖
